RIOT-OS/RIOT
# tests: add interactive_sync adapted to shell

### Contribution description

This PR changes the way Interactive Sync is done when the shell is present, instead of using new commands it prompts the shell with a `\n` until `>` is received as an answer

This will reduce the size of a lot of test applications. 

### Testing procedure

- Run all tests using shell on multiple boards (I'll run it on my setup as soon as the PR is open)

- Green murdock

- `tests/periph_wdt` and `tests/struc_tm_utility/` should still work:

```
PASSED target time: 128[ms], actual_time: 128.534[ms]
PASSED target time: 512[ms], actual_time: 502.025[ms]
PASSED target time: 1024[ms], actual_time: 1002.731[ms]
PASSED target time: 8192[ms], actual_time: 8001.558[ms]
TEST PASSED
```

```
BOARD=samr21-xpro make -C tests/struct_tm_utility/ flash test

>  day 2018 11 28
What weekday was 2018-11-28? The 332(th) day of the year was a WED.
day 2016 2 29
>  day 2016 2 29
What weekday was 2016-02-29? The 60(th) day of the year was a MON.
day 2017 2 29
>  day 2017 2 29
The supplied date is invalid, but no error should occur.
```

- tested with `miniterm`, `socat` and `pyterm`, can someone test for `rtt`?

### Issues/PRs references

Fixes compilation issues in #13152
